assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+# Knative Self-Assessment
+Security reviewers: Maya Humston, Ethan Vazquez, Zhikang Xu, Shambhavi Seth
+This document is intended to assess the current state of security for the Knative project.
+## Table of Contents
+* [Metadata](#metadata)
+  * [Software Bill of Materials](#software-bill-of-materials)
+  * [Security Links](#security-links)
+* [Overview](#overview)
+  * [Background](#background)
+  * [Actors](#actors)
+  * [Actions](#actions)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+  * [Deployment pipeline](#deployment-pipeline)
+  * [Communication channels](#communication-channels)
+  * [Ecosystem](#ecosystem)
+* [Security issue resolution](#security-issue-resolution)
+  * [Responsible disclosure practice](#responsible-disclosure-practice)
+  * [Incident Response](#incident-response)
+* [Appendix](#appendix)
+* [Recommendations](#recommendations)
+
+## Metadata
+| | |
+|-----------|------|
+| Software | https://github.com/knative/serving https://github.com/knative/eventing https://github.com/knative/func |
+| Security Provider? | No — the primary function of Knative is to provide a set of components and building blocks to extend Kubernetes functionality and should not be considered primarily a security provider. |
+| Languages | Go |
+| | |
```

Comment:
Please include if there an SBOM published by the knative team. if there isn't one, please note the same